src/mainmenu.asm

```diff
@@ -412,10 +412,28 @@ presets_goto_select_preset_category:
 
 presets_custom_preset_slot:
 if !FEATURE_TINYSTATES
-    %cm_numfield("Custom Preset Slot", !sram_custom_preset_slot, 0, 15, 1, 2, #0) ; update total slots in gamemode.asm
+    %cm_numfield("Custom Preset Slot", !sram_custom_preset_slot, 0, 15, 1, 2, #.routine) ; update total slots in gamemode.asm
 else
-    %cm_numfield("Custom Preset Slot", !sram_custom_preset_slot, 0, 39, 1, 2, #0) ; update total slots in gamemode.asm
+    %cm_numfield("Custom Preset Slot", !sram_custom_preset_slot, 0, 39, 1, 2, #.routine) ; update total slots in gamemode.asm
 endif
+  .routine
+    ; ignore if not A, X, or Y
+    LDA !IH_CONTROLLER_PRI_NEW : BIT #$40C0 : BNE .submenu
+    RTL
+  .submenu
+    LDA !sram_custom_preset_slot : BEQ .zero
```

Comment:
Should have put a comment here. Pressing A on the numfield will increment the slot before landing here, so we undo that before jumping to the hidden submenu.